client/starwhale/utils/venv.py

```diff
@@ -875,20 +875,20 @@ def install_starwhale(
     force: bool = False,
     configs: _ConfigsT = None,
 ) -> None:
-    if version == "" or version == SW_DEV_DUMMY_VERSION:
-        version = get_downloadable_sw_version()
-
-    req = SW_PYPI_PKG_NAME
-    if version:
-        req = f"{req}=={version}"
-
     _existed = check_user_python_pkg_exists(
         str(prefix_path / "bin" / "python3"), SW_PYPI_PKG_NAME
     )
     if _existed and not force:
         console.info(f"{SW_PYPI_PKG_NAME} has already be installed at {prefix_path}")
         return
 
+    if version == "" or version == SW_DEV_DUMMY_VERSION:
+        version = get_downloadable_sw_version()
+
+    req = [SW_PYPI_PKG_NAME]
+    if version:
+        req = [version] if re.match(r"^git", version) else [f"{req}=={version}"]
```

Comment:
`[f"{req}=={version}"]`  typo?

req is a list type.  add some ut?

Reply:
fixed